lib/ain-ocean/src/data_acces/masternode_states.rs

```diff
@@ -13,25 +14,71 @@ pub struct MasterStatsDb {
     pub db: RocksDB,
 }
 impl MasterStatsDb {
-    pub async fn get_latest(&self) -> Result<MasternodeStats> {
-        // let mut latest_stats: Option<MasternodeStats> = None;
-        // let mut highest_height = -1;
+    pub async fn get_latest(&self) -> Result<Option<MasternodeStats>> {
+        let latest_height_bytes = match self
+            .db
+            .get("masternode_block_height", b"master_block_height")
+        {
+            Ok(Some(value)) => value,
+            Ok(None) => return Ok(None), // No latest block height set
+            Err(e) => return Err(anyhow!(e)),
+        };
 
-        // let iter = self.db.iterator("masternode_stats", IteratorMode::End); // Start from the end of the DB
+        // Convert the latest height bytes back to an integer
+        let latest_height = i32::from_be_bytes(
+            latest_height_bytes
+                .as_slice()
+                .try_into()
+                .map_err(|_| anyhow!("Byte length mismatch for latest height"))?,
+        );
 
-        // for (key, value) in iter {
-        //     let stats: MasternodeStats = serde_json::from_slice(&value)?;
-        //     if stats.block.height > highest_height {
-        //         highest_height = stats.block.height;
-        //         latest_stats = Some(stats);
-        //     }
-        // }
-
-        // Ok(latest_stats);
-        todo!()
+        // Retrieve the block with the latest height
+        match self.db.get("block", &latest_height.to_be_bytes()) {
+            Ok(Some(value)) => {
+                let block: MasternodeStats =
+                    serde_json::from_slice(&value).map_err(|e| anyhow!(e))?;
+                Ok(Some(block))
+            }
+            Ok(None) => Ok(None), // No block found for the latest height
+            Err(e) => Err(anyhow!(e)),
+        }
     }
```

Comment:
Couldn't we iterator from End with limit = 1 and take the latest entry here ? This would bypass the need to query the block db as well